floss/language_identifier.py

```diff
@@ -149,3 +156,18 @@ def verify_pclntab(section, pclntab_va: int) -> bool:
         logger.error("Error parsing pclntab header")
         return False
     return True if pc_quanum in {1, 2, 4} and pointer_size in {4, 8} else False
+
+
+def is_dotnet_bin(pe: pefile.PE) -> bool:
+    """
+    Check if the binary is .net or not
+    Checks the IMAGE_DIRECTORY_ENTRY_COM_DESCRIPTOR entry in the OPTIONAL_HEADER of the file.
+    If the entry is not found, or if its size is 0, the file is not a .net file.
+    """
+    try:
+        directory_index = pefile.DIRECTORY_ENTRY["IMAGE_DIRECTORY_ENTRY_COM_DESCRIPTOR"]
+        dir_entry = pe.OPTIONAL_HEADER.DATA_DIRECTORY[directory_index]
+    except IndexError:
+        return False
+
+    return dir_entry.Size != 0
```

Comment:
not sure if this really improves things, but `dir_entry.Size` and `dir_entry.VirtualAddress` should be non zero.

Reply:
```suggestion
    return dir_entry.Size != 0 and dir_entry.VirtualAddress != 0
```